src/ext/default.c

```diff
@@ -26,41 +27,61 @@
 // normalize TF by number of tokens (weighted)
 #define NORM_DOCLEN 2
 
-// recursively calculate tf-idf
-static double tfidfRecursive(const RSIndexResult *r, const RSDocumentMetadata *dmd) {
+#define EXPLAIN(indexResult, fmt, args...)                                            \
```

Comment:
Why not using sds here?

Reply:
@mnunberg I will need an extra explanation for what you mean.